sema/subyt/sinks.py

```
@@ -136,15 +139,21 @@ def close(self):
     def _add(
         self, file_path: str, part: str, source_mtime: float | None = None
     ):
-        sink_mtime = (
-            Path(file_path).stat().st_mtime if Path(file_path).exists() else 0
-        )
+        out_path: Path = Path(file_path)
+        if out_path.exists() and out_path.is_dir():
+            log.warning(
+                f"Skipping creation of {file_path} as it is a directory. "
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)